[6.0] Formatting the attributes for Editor buttons.

karmendra opened the pull request:
The Editor buttons used for Datatables Editor when added to attribute list generates Javascript JSON code that has editor variable wrapped around with double quotes. which is treated as string in javascript instead of variable. This code change removes additionally added double quotes.

For more detailed information you can look at issue "Adding Editor buttons" #1095

yajra commented:
Is `editor` always the variable name for editor? Do not have license on editor and can't really test this one. Thanks!

karmendra commented:
You are right, it is not always editor, user can define there own variable names. See this link for sample code  (specially javascript) [https://editor.datatables.net/examples/inline-editing/simple.html](url).

Now I am not sure, how to easily handle user defined variables. 

Any pointer where I can look for solutions. Something like in getBuilderParameters datatables parameters I can add a param `'editor' =>[ 'variable'=>'editor1']`, and use that in Builder@parameterize function.

Or even better is, if I add a parameter `'editor' =>[ 'enabled' => true, 'ajax' => '../php/staf.php', 'table' =>'#example', 'fields'=>[['label'=>'First Name','name'=>'first_name'],['label'=>'Last Name','name'=>'last_name']]]`  it will generate the following editor script making sure the variable name is editor.

    editor = new $.fn.dataTable.Editor( {
        ajax: "../php/staff.php",
        table: "#example",
        fields: [ {
                label: "First name:",
                name: "first_name"
            }, {
                label: "Last name:",
                name: "last_name"
            }
        ]
    } );

PS: Sorry for late response, got busy with other projects.

karmendra commented:
Datatables Editor gives a 15 days trail license, if you haven't exhausted yours for your email id, you can register on datatables site and download the editor js.

yajra commented:
My trial license already expires. What I suggest is to make a dedicate decode method that allows plain script on editor key like on this line:

```php
$json = $this->decodeCallbackFunctions($callbackFunctions, $json);
```

karmendra commented:
I updated the code as you suggested, i added `encodeEditorButtons` and `decodeEditorButtons`, and it works like a charm. Let me know if you have more suggestions.

karmendra commented:
@yajra Anything else you want me to update with this PR? It will be really helpful for my project if this goes in. Thanks Again.

yajra commented:
Not really sure yet how to test this since I don't have editor license. Can I trust you on this one?

Anyone reading this can confirm this PR please? Thanks! :) 

Anyways, the PR looks good. Give me your confirmation that this works on your side and I think I will merge this. Thanks!

karmendra commented:
I tested it seems right to me, it is working fine on my Laravel 5.1 project.

yajra commented:
@karmendra trusting you on this one. Released on v6.28.1. 

BTW: can you also please submit the same PR on https://github.com/yajra/laravel-datatables-html for v7.0 support? Thanks a lot 👍
